diff --git a/poppler/Object.h b/poppler/Object.h
--- a/poppler/Object.h
+++ b/poppler/Object.h
@@ -234,5 +234,5 @@ public:
     explicit Object(long long int64gA) : type { objInt64 }, data { int64gA } { }
-
+
-    explicit Object(std::unique_ptr<Array> arrayA) : type { objArray }, data { std::shared_ptr<Array>(std::move(arrayA)) } { }
+    explicit Object(std::unique_ptr<Array> arrayA);
-
+
     explicit Object(std::unique_ptr<Dict> &&dictA) : type { objDict }, data { std::shared_ptr<Dict>(std::move(dictA)) } { }
diff --git a/poppler/Object.cc b/poppler/Object.cc
--- a/poppler/Object.cc
+++ b/poppler/Object.cc
@@ -41,5 +41,7 @@
 static const char *objTypeNames[numObjTypes] = { "boolean", "integer", "real", "string", "name", "null", "array", "dictionary", "stream", "ref", "cmd", "error", "eof", "none", "integer64", "hexString", "dead" };
-
+
+Object::Object(std::unique_ptr<Array> arrayA) : type { objArray }, data { std::shared_ptr<Array>(std::move(arrayA)) } { }
+
 Object Object::copy() const
 {
     CHECK_NOT_DEAD;
